scripts/install_xcpng.py

```diff
@@ -0,0 +1,225 @@
+        return is_uuid(output)
+    except Exception:
+        return False
+
+def check_mac_address(host, mac_address):
+    bridge = host.inventory['MANAGEMENT_INTERFACE']
+    host_mac_address = host.ssh(['cat', f'/sys/class/net/{bridge}/address'])
+    if mac_address != host_mac_address:
+        raise Exception(
+            f"Unexpected MAC address {host_mac_address} for host {host.hostname_or_ip}. Expected: {mac_address}"
+        )
+
+def url_checker(url):
+    try:
+        response = requests.get(url)
+        if not response:
+            raise Exception(f"{url}: URL is not reachable, status_code: {response.status_code}")
+    except requests.exceptions.RequestException as e:
+        raise SystemExit(f"{url}: URL is not reachable\nErr: {e}")
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("host", help="The IP address or the name of the host where the VM is created.")
+    parser.add_argument("vm_uuid", help="The UUID of an existing VM to use to create the new host.")
+    parser.add_argument("action", metavar='action', choices=['install', 'upgrade'], help="install or upgrade.")
+    parser.add_argument("version_xcpng", help="The version of xcp-ng we wish.")
+    parser.add_argument("--installer", help="URL of the installer on the PXE")
+    args = parser.parse_args()
+
+    try:
+        is_uuid(args.vm_uuid)
+    except Exception:
+        raise Exception(f'The VM UUID given is invalid: {args.vm_uuid}')
+
+    if args.version_xcpng[0].isdigit():
+        version1 = args.version_xcpng
+    else:
+        raise Exception(f'The version does not seem valid: {args.version_xcpng}')
+
+    if not args.installer:
+        pxe = pxe_address()
+        installer = f"http://{pxe}/installers/xcp-ng/{version1}/"
+    else:
+        installer = args.installer
+
+    url_checker(f"{installer}.treeinfo")
+
+    pool = Pool(args.host)
+    host = pool.master
+    vm = VM(args.vm_uuid, host)
+    vif = vm.vifs()[0]
+    mac_address = vif.param_get('MAC')
+    with tempfile.TemporaryDirectory(suffix=mac_address, ignore_cleanup_errors=True) as tmp_local_path:
+        print('generate files: answerfile.xml and boot.conf')
+        if vm.is_uefi:
+            hdd = 'nvme0n1'
+        else:
+            hdd = 'sda'
+        generate_answerfile(tmp_local_path, installer, args.host, args.action, hdd)
+        generate_boot_conf(tmp_local_path, installer)
+        print('Copy files to the pxe server')
+        copy_files_to_pxe(mac_address, tmp_local_path)
+        atexit.register(lambda: clean_files_on_pxe(mac_address))
+        print('Start the vm to install')
+        vm.start()
+        time.sleep(20)
+        clean_bootconf_on_pxe(mac_address)
+        print("Waiting for the installation, and for the host to be ready.")
+        wait_for(
+            lambda: get_new_host_ip(mac_address) is not None, "Waiting for the new host to installed and up", 3600, 10
+        )
```

Comment:
The first print is redundant with the one in the wait call

Reply:
Actually wait_for doesn't display anything because it uses logging. Maybe there's a log level we need to tweak?

Reply:
hmm IDK